src/libstd/net/ip.rs

```diff
@@ -124,13 +116,21 @@ pub struct Ipv6Addr {
 
 #[allow(missing_docs)]
```

Comment:
I just noticed that the `Reserved` and `Unassigned` variants are part of the table but are missing from the enum. Instead, `Ipv6Address::multicast_scope()` returns `None` for such scopes. I don't think this is a problem but I think it's worth mentioning. Maybe @therealbstern or @the8472 have an opinion on that API?

Reply:
FWIW, I think Reserved and Unassigned are good things to put in the enum, but my concern is that there's a non-zero chance the Reserved/Unassigned space will get assigned to something in the future.

On the other hand, `None` is just as wrong as `Reserved` or `Unassigned` if the IETF assigns them in the future.  In addition, IPv6 is so big that who knows when/if they'll get assigned.

My measured opinion is that we should add `Reserved` and `Unassigned` right now, since it's correct right now and might remain correct indefinitely.

Reply:
I think it’s something to consider now before stabilization.

Let’s at least drop a `#[non_exhaustive]` attribute on the `enum` so we can add those variants later if need be.

Reply:
Yes, nonexhaustive should cover future RFC changes nicely.

Reply:
sounds good, I'll make the changes.